feat(api): add baseline labware detection method to flex stacker driver.

vegano1 opened the pull request:
# Overview

Implement the `baseline` labware detection method for now so we can support error recovery flows, we might change this or add a different labware detection method later

## Test Plan and Hands-on Testing

- [x] Make sure we can detect labware on the Z-axis Extended/Retracted position
- [x] Make sure we can detect labware on the X-axis retracted position.

## Changelog

- add the `baseline` labware detection method to the TOF sensor driver

## Review requests

- Does this make sense?

## Risk assessment

Low, unreleased

vegano1 committed:
feat(api): add baseline labware detection method to flex stacker driver.

sfoster1 reviewed (approved):
Can we put the json file almost anywhere else? shared-data, hardware... I really don't like having it in api since it's so massive. It seems like it will all work, though.

ahiuchingau reviewed (commented):
Agreed with Seth that the baseline should be stored in shared-module-data. 

If this is blocking testing, feel free to merge now but please make sure to put these TOF config and baseline data in module shared-data in a follow-up PR.

vegano1 committed:
Merge branch 'edge' into flex-stacker-add-basic-tof-labware-detection

vegano1 committed:
move the tof_baseline.json file to opentrons_shared_data/module/

vegano1 committed:
format

vegano1 committed:
add uniqueModuleData to module definition to keep data only the module cares about
- use the uniqueModuleData field to keep track of tof sensor baseline values for labware detection

vegano1 committed:
move labware_detected to the FlexStacker module instead of the driver
move load_tof_baseline_data from the driver to shared-data

vegano1 committed:
Merge branch 'edge' into flex-stacker-add-basic-tof-labware-detection

vegano1 committed:
create new tof baseline to include dvt units

ahiuchingau commented on api/src/opentrons/hardware_control/modules/flex_stacker.py:
```diff
@@ -68,6 +71,9 @@
 MEDIUM_LABWARE_Z_LIMIT = 20.0
 
+# The name of the tof sensor baseline file
+TOF_BASELINE_FILE = "tof_baseline.json"
```
This line can be removed

ahiuchingau reviewed (approved):
FANTASTIC!!

vegano1 committed:
remove unused constant
